nautobot/extras/filters.py

```diff
@@ -98,6 +100,129 @@ class CreatedUpdatedFilterSet(django_filters.FilterSet):
     last_updated__lte = django_filters.DateTimeFilter(field_name="last_updated", lookup_expr="lte")
 
 
+class RelationshipFilter(django_filters.MultipleChoiceFilter):
+    """
+    Filter objects by the presence of associations on a given Relationship.
+    """
+
+    def __init__(self, side, relationship=None, queryset=None, choices=None, *args, **kwargs):
+        self.relationship = relationship
+        self.qs = queryset
+        self.side = side
+        super().__init__(choices=choices, *args, **kwargs)
+
+    def filter(self, qs, value):
+        # Check if value is empty or a DynamicChoiceField that is empty.
+        if not value or "" in value:
+            # if value is empty we return the entire unmodified queryset
+            return qs
+        else:
+            if self.side == "source":
+                values = RelationshipAssociation.objects.filter(
+                    destination_id__in=value,
+                    source_type=self.relationship.source_type,
+                    relationship=self.relationship,
+                ).values_list("source_id", flat=True)
+            elif self.side == "destination":
+                values = RelationshipAssociation.objects.filter(
+                    source_id__in=value,
+                    destination_type=self.relationship.destination_type,
+                    relationship=self.relationship,
+                ).values_list("destination_id", flat=True)
+            else:
+                destinations = RelationshipAssociation.objects.filter(
+                    source_id__in=value,
+                    destination_type=self.relationship.destination_type,
+                    relationship=self.relationship,
+                ).values_list("destination_id", flat=True)
+
+                sources = RelationshipAssociation.objects.filter(
+                    destination_id__in=value,
+                    source_type=self.relationship.source_type,
+                    relationship=self.relationship,
+                ).values_list("source_id", flat=True)
+
+                values = list(chain(destinations, sources))
```

Comment:
Is this different/more efficient than `values = list(destinations) + list(sources)`?